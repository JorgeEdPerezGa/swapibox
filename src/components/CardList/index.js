import React, { Component } from 'react';
import PeopleCard from '../Cards/PeopleCard';
import PlanetCard from '../Cards/PlanetCard';
import VehicleCard from '../Cards/VehicleCard';

class CardList extends Component {
  constructor() {
    super();
    this.state = {};
  }

  componentWillMount() {
    this.setState({ favorites: [] });
  }

  handleFavorites = (card) => {
    const favorites = this.state.favorites;
    const removeFavorite = favorites.filter(cards => cards !== card);
    const newFavorites = [...favorites, card];
    if (favorites.includes(card)){
      return this.setState({ favorites: removeFavorite });
    }
    this.setState({ favorites: newFavorites });
  }

  render() {
    console.log(this.state.favorites);
    if (this.props.type && this.props.title === 'people') {
      let cards = this.props.type.results.map( (card, index) => {
        return (
          <PeopleCard
            key = {`${index}${card.name}`}
            card = { card }
            name = { card.name }
            birthYear = { card.birthYear }
            eyeColor = { card.eyeColor }
            skinColor = { card.skinColor }
            height = { card.height }
            mass = { card.mass }
            homeworld = { card.homeworld }
            homeworldPopulation = { card.homeworldPopulation }
            handleFavorites = { this.handleFavorites }/>
        );
      });

      return (
        <section className = {`card-list-${this.props.title}`}>
          { cards }
        </section>
      );
    }

    if (this.props.type && this.props.title === 'planets') {
      let cards = this.props.type.results.map( (card, index) => {
        return (
          <PlanetCard
            key = {`${index}${card.name}`}
            name = { card.name }
            climate = { card.climate }
            population = { card.population }
            residents = { card.residents }/>
        );
      });

      return (
        <section className = {`card-list-${this.props.title}`}>
          { cards }
        </section>
      );
    }

    if (this.props.type && this.props.title === 'vehicles') {
      let cards = this.props.type.results.map( (card, index) => {
        console.log(card);
        return (
          <VehicleCard
            key = {`${index}${card.name}`}
            name = { card.name }
            model = { card.model }
            vehicleClass = { card.vehicleClass }
            passengers = { card.passengers }
            manufacturer = { card.manufacturer }/>
        );
      });

      return (
        <section className = {`card-list-${this.props.title}`}>
          { cards }
        </section>
      );
    }

    return (
      <div>loading</div>
    );
  }
}

export default CardList;
