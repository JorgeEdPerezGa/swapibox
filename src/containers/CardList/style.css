.card-list {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  padding-top: 20px; }
