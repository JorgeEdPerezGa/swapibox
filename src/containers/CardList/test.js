import React from 'react';
import ReactDOM from 'react-dom';
import CardList from './index';
import setupTest from '../../setupTests';
import { shallow } from 'enzyme';

describe('<CardList/>', () => {
  const wrapper = shallow(<CardList />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
