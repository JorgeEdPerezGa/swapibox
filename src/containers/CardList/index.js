import React, { Component } from 'react';
import PeopleCard from '../../components/Cards/PeopleCard';
import PlanetCard from '../../components/Cards/PlanetCard';
import VehicleCard from '../../components/Cards/VehicleCard';
import './style.css';

class CardList extends Component {
  constructor() {
    super();
    this.state = {};
  }

  componentWillMount() {
    this.setState({ favorites: [] });
  }

  handleFavorites = (card) => {
    const favorites = this.state.favorites;
    const removeFavorite = favorites.filter(cards => cards !== card);
    const newFavorites = [...favorites, card];
    if (favorites.includes(card)){
      return this.setState({ favorites: removeFavorite });
    }
    this.setState({ favorites: newFavorites });
  }

  renderCards = () => {
    if (this.props.type) {
      let cards = this.props.type.results.map( (card, index) => {

        if (this.props.type && this.props.title === 'people') {
          return (
            <PeopleCard
              key = {`${index}${card.name}`}
              card = { card }
              name = { card.name }
              birthYear = { card.birthYear }
              eyeColor = { card.eyeColor }
              skinColor = { card.skinColor }
              height = { card.height }
              mass = { card.mass }
              species = { card.species }
              homeworld = { card.homeworld }
              homeworldPopulation = { card.homeworldPopulation }
              handleFavorites = { this.handleFavorites }/>
          );
        }

        if (this.props.type && this.props.title === 'planets') {
          return (
            <PlanetCard
              key = {`${index}${card.name}`}
              card = { card }
              name = { card.name }
              climate = { card.climate }
              population = { card.population }
              residents = { card.residents }
              handleFavorites = { this.handleFavorites }/>
          );
        }

        if (this.props.type && this.props.title === 'vehicles') {
          return (
            <VehicleCard
              key = {`${index}${card.name}`}
              card = { card }
              name = { card.name }
              model = { card.model }
              vehicleClass = { card.vehicleClass }
              passengers = { card.passengers }
              manufacturer = { card.manufacturer }
              handleFavorites = { this.handleFavorites }/>
          );
        }

      });

      return (
        <section className = {`card-list card-list-${this.props.title}`}>
          { cards }
        </section>
      );
    } else {
      return ( <div>loading</div> );
    }
  }

  render() {
    console.log(this.state.favorites);
    return (
      <div className = 'card-list'>
        {this.renderCards()}
      </div>
    );
  }
}

export default CardList;
