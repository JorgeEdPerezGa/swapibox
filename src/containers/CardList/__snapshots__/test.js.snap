// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList/> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <img
                height="200px"
                src="https://cdnb.artstation.com/p/assets/images/images/003/827/899/original/matthew-garwin-loading-vector.gif?1477723699"
                width="200px"
        />
</div>,
      "className": "card-list",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          height="200px"
          src="https://cdnb.artstation.com/p/assets/images/images/003/827/899/original/matthew-garwin-loading-vector.gif?1477723699"
          width="200px"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "200px",
          "src": "https://cdnb.artstation.com/p/assets/images/images/003/827/899/original/matthew-garwin-loading-vector.gif?1477723699",
          "width": "200px",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <img
                    height="200px"
                    src="https://cdnb.artstation.com/p/assets/images/images/003/827/899/original/matthew-garwin-loading-vector.gif?1477723699"
                    width="200px"
          />
</div>,
        "className": "card-list",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            height="200px"
            src="https://cdnb.artstation.com/p/assets/images/images/003/827/899/original/matthew-garwin-loading-vector.gif?1477723699"
            width="200px"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "200px",
            "src": "https://cdnb.artstation.com/p/assets/images/images/003/827/899/original/matthew-garwin-loading-vector.gif?1477723699",
            "width": "200px",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
